bin/proving-service/build.rs

```diff
@@ -0,0 +1,67 @@
+use std::{
+    env, fs,
+    path::{Path, PathBuf},
+};
+
+use miette::IntoDiagnostic;
+use protox::prost::Message;
+
+/// Defines whether the build script can write to /src.
+const CAN_WRITE_TO_SRC: bool = option_env!("DOCS_RS").is_none();
+
+/// Generates Rust protobuf bindings from .proto files.
+fn main() -> miette::Result<()> {
+    // The docs.rs build pipeline has a read-only filesystem, so we want to return early or
+    // otherwise the docs will fail to build there.
+    if !CAN_WRITE_TO_SRC {
+        return Ok(());
+    }
+
+    compile_tonic_server_proto()
+}
+fn compile_tonic_server_proto() -> miette::Result<()> {
+    let crate_root =
+        PathBuf::from(env::var("CARGO_MANIFEST_DIR").expect("CARGO_MANIFEST_DIR should be set"));
+    let dst_dir = crate_root.join("src").join("generated");
+
+    // Construct the path to the miden-remote-provers/proto/api.proto file
+    let proto_dir = crate_root
+        .parent() // Go up to the workspace root
+        .expect("bin directory should exist")
+        .parent() // Go up to the workspace root
+        .expect("Workspace root should exist")
+        .join("miden-remote-provers")
+        .join("proto");
```

Comment:
I thinl, ideally, the proto definitions should be defined in a common, top-level directory if possible, as done in the `miden-node` and as suggested [here](https://github.com/0xPolygonMiden/miden-base/issues/975#:~:text=we%20can%20have%20a%20top%2Dlevel%20directory%20for%20protobuf%20files%20and%20these%20would%20be%20copied%20during%20build%20time%20to%20the%20required%20crates%20(similar%20to%20how%20we%20do%20it%20in%20miden%2Dnode).).